docs/source/redirects.py

```
@@ -55,12 +90,45 @@ def generate_redirects(app):
+            % (base_page, len(anchor_map))
+        )
+
+        redirect_path = os.path.join(app.builder.outdir, base_page)
+        etau.write_file(
+            _HASH_REDIRECT_TEMPLATE.format(anchor_map=anchor_map_json),
+            redirect_path,
+        )
+
+
+def generate_redirects(app):
+    if not type(app.builder) == builders.StandaloneHTMLBuilder:
```

Ruff (0.14.0)

Use `is` and `is not` for type comparisons, or `isinstance()` for isinstance checks

(E721)